src/commands/infra/generate.ts

```diff
@@ -0,0 +1,190 @@
+      logger.error(
+        `Provided project path for generate is invalid or definition.json can not be found: ${projectPath}`
+      );
+      return false;
+    }
+    logger.info(
+      `Project folder found. Attempting to generate definition.json file.`
+    );
+  } catch (_) {
+    logger.error(`Unable to validate project folder path.`);
+    return false;
+  }
+  return true;
+};
+
+/**
+ * Checks if working definition.json is present in the provided project path with validated source & version
+ *
+ * @param projectPath Path to the definition.json file
+ */
+export const validateTemplateSource = async (
+  projectPath: string
+): Promise<string[]> => {
+  try {
+    const rootDef = path.join(projectPath, "definition.json");
+    const data: string = fs.readFileSync(rootDef, "utf8");
+    const definitionJSON = JSON.parse(data);
+    // TO DO : Check for malformed JSON
+    if (!(definitionJSON.template && definitionJSON.source)) {
+      logger.info(
+        `The definition.json file is invalid. There is a missing field for the definition file's sources. Template: ${definitionJSON.template} source: ${definitionJSON.source} version: ${definitionJSON.version}`
+      );
+      return [];
+    }
+    logger.info(
+      `Checking for locally stored template: ${definitionJSON.template} from remote repository: ${definitionJSON.source} at version: ${definitionJSON.version}`
+    );
+    const sources = [
+      definitionJSON.source,
+      definitionJSON.template,
+      definitionJSON.version
+    ];
+    return sources;
+  } catch (_) {
+    logger.error(
+      `Unable to validate project folder definition.json file. Is it malformed?`
+    );
+    return [];
+  }
+  return [];
+};
+
+/**
+ * Checks if provided source, template and version are valid. TODO/ Private Repo, PAT, ssh-key agent
+ *
+ * @param projectPath Path to the definition.json file
+ */
+export const validateRemoteSource = async (
+  definitionJSON: string[]
+): Promise<boolean> => {
+  try {
+    const [source, template, version] = definitionJSON;
+    // Converting source name to storable folder name
+    const httpReg = /^(.*?)\.com/;
+    const punctuationReg = /[^\w\s]/g;
+    let sourceFolder = source.replace(httpReg, "");
+    sourceFolder = sourceFolder.replace(punctuationReg, "_").toLowerCase();
+    const sourcePath = path.join(spkTemplatesPath, sourceFolder);
+    logger.warn(`Converted to: ${sourceFolder}`);
+    logger.info(`Checking if source:${sourcePath} is stored locally.`);
+    if (!fs.existsSync(sourcePath)) {
+      logger.warn(
+        `Provided source template folder was not found, attempting to clone the template source repo locally.`
+      );
+      fs.mkdirSync(sourcePath);
+    } else {
+      logger.info(
+        `Source template folder found. Checking remote existence of remote repository`
+      );
+    }
+    // Checking for git remote
+    require("simple-git")(sourcePath).listRemote(
```

Comment:
The require is done once for "simpleGit" which is instantiated with "git = simpleGit()", why the use of require here?  Is it possible to not do the require multiple times?

Reply:
Simple-git typescript definitions are constructed weird where the `sourcePath` maps to where I want the git command to run. The alternative would be to change the working path every time to the respective folder `process.chdir(${sourcePath)` It would still have redundancy. Open to suggestions.

Reply:
Keep as is.

Reply:
does having `import simpleGit from "simple-git"` on top doing `simpleGit(sourcePath)` not work?